javatests/arcs/sdk/HandleHolderBaseTest.kt

```diff
@@ -0,0 +1,303 @@
+    val handleHolder = HandleHolderBase(
+      "TestParticle",
+      entitySpecs = mapOf("readHandle" to mock())
+    )
+
+    assertFailsWith(
+      NoSuchElementException::class,
+      "Particle TestParticle does not have a handle with name handleButNotEntities."
+    ) {
+      handleHolder.setHandle("handleButNotEntities", mock())
+    }
+  }
+
+  @Test
+  fun setHandle_handelNameInSpec_success() {
+    class DummyHandle : HandleHolderBase(
+      "TestParticle",
+      entitySpecs = mapOf("writeHandle" to mock())
+    ) {
+      val writeHandle: Handle by handles
+    }
+
+    val handleHolder = DummyHandle()
+    val mock = mock<Handle>()
+
+    handleHolder.setHandle("writeHandle", mock)
+
+    assertThat(handleHolder.writeHandle).isSameInstanceAs(mock)
+  }
+
+  @Test
+  fun detach_allHandles_getUnregistered() {
+    val handleHolder = HandleHolderBase(
+      "TestParticle",
+      entitySpecs = mapOf(
+        "readHandle" to mock(),
+        "writeHandle" to mock()
+      )
+    )
+    val readMock = mock<Handle>()
+    val writeMock = mock<Handle>()
+    handleHolder.setHandle("readHandle", readMock)
+    handleHolder.setHandle("writeHandle", writeMock)
+
+    handleHolder.detach()
+
+    verify(readMock, times(1)).unregisterForStorageEvents()
+    verify(writeMock, times(1)).unregisterForStorageEvents()
+  }
+
+  @Test
+  fun reset_allHandles_getUnregistered() {
+    val handleHolder = HandleHolderBase(
+      "TestParticle",
+      entitySpecs = mapOf(
+        "readHandle" to mock(),
+        "writeHandle" to mock()
+      )
+    )
+    val readMock = mock<Handle>()
+    val writeMock = mock<Handle>()
+    handleHolder.setHandle("readHandle", readMock)
+    handleHolder.setHandle("writeHandle", writeMock)
+
+    handleHolder.reset()
+
+    verify(readMock, times(1)).unregisterForStorageEvents()
+    verify(writeMock, times(1)).unregisterForStorageEvents()
+  }
+
+  @Test
+  fun reset_allHandles_getClosed() {
+    val handleHolder = HandleHolderBase(
+      "TestParticle",
+      entitySpecs = mapOf(
+        "readHandle" to mock(),
+        "writeHandle" to mock()
+      )
+    )
+    val readMock = mock<Handle>()
+    val writeMock = mock<Handle>()
+    handleHolder.setHandle("readHandle", readMock)
+    handleHolder.setHandle("writeHandle", writeMock)
+
+    handleHolder.reset()
+
+    verify(readMock, times(1)).close()
+    verify(writeMock, times(1)).close()
+  }
+
+  @Test
+  fun reset_allHandles_allElementsRemoved() {
+    val handleHolder = HandleHolderBase(
+      "TestParticle",
+      entitySpecs = mapOf(
+        "readHandle" to mock(),
+        "writeHandle" to mock()
+      )
+    )
+    handleHolder.setHandle("readHandle", mock())
+    handleHolder.setHandle("writeHandle", mock())
+    assertThat(handleHolder.handles).isNotEmpty()
+
+    handleHolder.reset()
+
+    assertThat(handleHolder.handles).isEmpty()
+  }
+
+  @Test
+  fun isEmpty_delegatesToHandlesMap_returnsTrue() {
```

Comment:
I'm not sure what `delegatesToHandlesMap` means, it think it is more confusing than helpful

Reply:
yeah, those don't make too much sense.